src/components/users/user.ts

```diff
@@ -2,12 +2,19 @@ import { Model } from 'objection'
 
 import { Group } from '../groups/group'
 
+export enum RoleType { 
+    ADMIN = 'ADMIN',
+    TICKET_ADMIN = 'TICKET_ADMIN',
+    USER = 'USER'
+}
+
+
 export class User extends Model {
```

Comment:
I'm thinking about adding a method for the User class that can be used to check it's role, similar to what I suggested in my other comment in `config/passport.ts`. What do you think about this approach?

Reply:
I like the middleware approach. What kind of implementation are you thinking?

`(roles: RoleType[]) => {
\treturn roles.some(element =>  role == element)
}`

or

`isAdmin: () => role == RoleType.ADMIN `

Reply:
I thought about two middlewares like this:

`requireAnyRoles: () => `

`requireAllRoles: () =>`

Reply:
> I like the middleware approach. What kind of implementation are you thinking?
> 
> `(roles: RoleType[]) => roles.some(element => role == element) `
> 
> or
> 
> `isAdmin: () => role == RoleType.ADMIN `

I think the first approach is more flexible, so I'd go for that.